lib/rspec-puppet/monkey_patches.rb

```diff
@@ -1,5 +1,12 @@
 require 'pathname'
 
+# Load this library before enabling the monkey-patches to avoid HI-581
+begin
+require 'hiera/util/win32'
```

Comment:
Hiera 3 is basically EOL, so this path shouldn't change.

But playing the load order game is a bit like playing roulette.  I'd rather see any mocking of `FILE::ALT_SEPARATOR` removed from `rspec-puppet`. Otherwise, this will be a constant game of whack-a-mole any time code is discovered that correctly uses `FILE::ALT_SEPARATOR` to guard execution on Windows.

Reply:
@Iristyle I agree that having to mock these constants is undesirable, but it is an unfortunate necessity at the moment. Long term I'd like to get rid of them, but for now we have to work with what we've got.